### ISSUE TITLE
[Bug]: Module `logical`: `all()`, `any()` must return 0-dim array

### ISSUE THREAD

**Issue by ClaudiaComito** [helmholtz-analytics/heat#1194]
### What happened?

Following the Python Array API, logical functions `all()` and `any()`  should return a zero-dimensional array with the test result "if a logical AND reduction was performed over the entire array" (cf. [`all()`](https://data-apis.org/array-api/2022.12/API_specification/generated/array_api.all.html?highlight=logical#api-specification-generated-array-api-all--page-root))

`ht.all()` and `ht.any()` return a 1-dim array instead (see below).

### Code snippet triggering the error

```shell
>>> import heat as ht
>>> a = ht.arange(10)
>>> b = ht.arange(10)
>>> (a==b).all()
DNDarray([True], dtype=ht.bool, device=cpu:0, split=None)
>>> (a==b).all().ndim
1

>>> import numpy as np
>>> (a.numpy() == b.numpy()).all()
True
>>> (a.numpy() == b.numpy()).all().ndim
0

>>> import torch
>>> a = torch.arange(10)
>>> b = torch.arange(10)
>>> (a==b).all()
tensor(True)
>>> (a==b).all().ndim
0
```

### Version

main (development branch)

### Python version

3.10

### PyTorch version

1.12

### MPI version

```shell
doesn't matter
```

**Comment by ClaudiaComito** [comment]
This issue is a duplicate of #1004. Will be fixed by #1005 . Closing.
